File: Lyd.MoopDemo/Lyd.MoopDemo.PmsConsole/PmsConsoleApp.cs
using System;
using System.Collections.Generic;
using Lyd.MoopDemo.PmsCore;

namespace Lyd.MoopDemo.PmsConsole
{
    public class PmsConsoleApp
    {
        private PmsApp pmsApp;

        public PmsConsoleApp()
        {
            pmsApp = new PmsApp();
            this.Run();
        }

        private void Run()
        {
            pmsApp.InitGuests();
            this.ShowMenuScreen();
        }

        private void ShowMenuScreen()
        {
            Console.Clear();
            Console.WriteLine("PHAN MEM QUAN LY KHACH SAN");
            Console.WriteLine("1. Quan ly ho so khach");
            Console.WriteLine("2. Quan ly don dat phong");
            Console.WriteLine("3. Quan ly checkin/checkout");
            Console.WriteLine("0. Thoat");
            Console.Write("Chon: ");
            int choice = int.Parse(Console.ReadLine());
            switch (choice)
            {
                case 1:
                    ShowManageGuestProfilesScreen();
                    break;
                case 2:
                    //TODO
                    ShowMenuScreen();
                    break;
                case 3:
                    //TODO
                    ShowMenuScreen();
                    break;
                case 0:
                    Console.WriteLine("Cam on va hen gap lai!");
                    break;
                default:
                    ShowMenuScreen();
                    break;
            }
        }

        private void ShowManageGuestProfilesScreen()
        {
            List<Guest> guests = pmsApp.GetGuests();
            Console.Clear();
            Console.WriteLine($"QUAN LY HO SO KHACH\nMa\tTen");
            ShowGuests(guests);

            Console.WriteLine("1. Tim kiem ho so khach");
            Console.WriteLine("2. Them ho so khach moi");
            Console.WriteLine("0. Quay lai");
            Console.Write("Chon: ");
            int choice = int.Parse(Console.ReadLine());
            switch (choice)
            {
                case 1:
                    ShowSearchGuestsScreen();
                    break;
                case 2:
                    ShowCreateGuestScreen();
                    break;
                case 0:
                    ShowMenuScreen();
                    break;
                default:
                    ShowManageGuestProfilesScreen();
                    break;
            }
        }

        private void ShowSearchGuestsScreen()
        {
            Console.Clear();
            Console.WriteLine($"TIM KIEM HO SO KHACH");
            Console.Write("Nhap ten: ");
            string name = Console.ReadLine();
            List<Guest> guests = pmsApp.SearchGuest(new SearchOption(name));

            if (guests.Count == 0)
            {
                Console.WriteLine("Khong co ket qua.");
                Console.WriteLine("1. Tiep tuc tim kiem");
                Console.WriteLine("0. Quay lai");
                Console.Write("Chon: ");
                int choice = int.Parse(Console.ReadLine());
                switch (choice)
                {
                    case 1:
                        ShowSearchGuestsScreen();
                        break;
                    case 0:
                        ShowManageGuestProfilesScreen();
                        break;
                    default:
                        break;
                }
            }
            else
            {
                ShowGuests(guests);
                Console.Read();
                ShowManageGuestProfilesScreen();
            }

           
        }

        private void ShowCreateGuestScreen()
        {
            Console.Clear();
            Console.WriteLine($"THEM HO SO KHACH");
            Console.Write("Nhap ten: ");
            string name = Console.ReadLine();

            Console.Write("Nhap so dien thoai: ");
            string phone = Console.ReadLine();

            Console.Write("Nhap dia chi: ");
            string address = Console.ReadLine();

            Console.Write("Nhap gioi tinh (Nam: M, Nu: F):");
            string genderText = Console.ReadLine();
            Gender gender = genderText == "M" ? Gender.Male : Gender.Female;

            var createdGuest = pmsApp.CreateGuest(name, address, phone, gender);
            Console.WriteLine($"Da them thanh cong. Ma khach : {createdGuest.GuestCode}");
            Console.Read();
            this.ShowManageGuestProfilesScreen();
        }

        private void ShowGuests(List<Guest> guests)
        {
            foreach (var guest in guests)
            {
                Console.WriteLine($"{guest.GuestCode}\t{guest.GuestName}");
            }
        }





    }
}

File: Lyd.MoopDemo_Java/pmsconsole/src/main/java/lyd/demo/pmsconsole/PmsConsoleApp.java
package lyd.demo.pmsconsole;

import lyd.demo.pmscore.Guest;
import lyd.demo.pmscore.PmsApp;

import java.util.List;
import java.util.Scanner;

public class PmsConsoleApp {
    private PmsApp app;

    public PmsConsoleApp() {
        app = new PmsApp();
        this.run();
    }

    public void run() {
        this.app.initGuests();
        this.selectMainMenu();
    }

    private void clearScreen() {
        System.out.print("\033[H\033[2J");
        System.out.flush();
    }

    private void selectMainMenu() {

        int selection;
        Scanner input = new Scanner(System.in);

        /***************************************************/
        this.clearScreen();
        System.out.println("PHAN MEM QUAN LY KHACH SAN");
        System.out.println("-------------------------\n");
        System.out.println("1. Quan ly ho so khach");
        System.out.println("2. Quan ly don dat phong");
        System.out.println("3. Quan ly checkin/checkout");
        System.out.println("0. Thoat");

        selection = input.nextInt();
        switch(selection){
            case 1:
                selectManageGuestMenu();
                break;
            case 2:
                //TO BE IMPLEMENTED
                break;
            case 3:
                //TO BE IMPLEMENTED
                break;
            case 0:
                break;
        }
    }

    private void selectManageGuestMenu() {

        int selection;
        Scanner input = new Scanner(System.in);

        /***************************************************/
        this.clearScreen();
        System.out.println("QUAN LY KHACH");
        System.out.println("-------------------------\n");

        this.showGuests(this.app.getGuests());

        /***************************************************/
        System.out.println("1. Tim kiem ho so khach");
        System.out.println("2. Them ho so khach");
        System.out.println("0. Thoat");

        selection = input.nextInt();
        switch(selection){
            case 1:
                this.searchGuests();
                break;
            case 2:
                //TO BE IMPLEMENTED
                break;
            case 3:
                //TO BE IMPLEMENTED
                break;
            case 0:
                this.selectMainMenu();
                break;
        }
    }

    private void showGuests(List<Guest> guests) {
        System.out.println("Ma\tTen");
        guests.stream().forEach(g -> {
            System.out.printf("%s\t%s\n", g.getGuestCode(), g.getGuestName());
        });
    }

    private void searchGuests(){
        String searchKey;
        Scanner input = new Scanner(System.in);

        /***************************************************/
        this.clearScreen();
        System.out.println("TIM KIEM KHACH");
        System.out.println("-------------------------\n");
        System.out.println("Nhap ten khach");

        searchKey = input.nextLine();
        List<Guest> guests = this.app.searchGuests(searchKey);
        if (guests.size() > 0) {
            this.showGuests(guests);
            input.nextLine();
            this.selectManageGuestMenu();
        } else {
            System.out.println("Khong co ket qua");
            System.out.println("-------------------------\n");
            System.out.println("1. Tim kiem lai");
            System.out.println("0. Quay lai");
            int selection = input.nextInt();
            switch(selection){
                case 1:
                    this.searchGuests();
                    break;
                case 0:
                    this.selectManageGuestMenu();
                    break;
            }
        }


    }
}
